SConstruct

```diff
@@ -733,6 +741,31 @@ def should_prepare_targets(style, toolchain, variant):
         if should_prepare_target(t, style, toolchain, variant):
             return True
 
+def should_build_ninja(style, toolchain, variant):
+    """
+    Return True if a ninja build file should be generated.
+
+    Typically, scons will be called as follows to generate a ninja build file:
+    `scons ninja=1 gcc.debug` where `gcc.debug` may be replaced with any of our
+    non-visual studio targets. Raise an exception if we cannot generate the
+    requested ninja build file (for example, if multiple targets are requested).
+    """
+    if 'ninja' not in ARGUMENTS or ARGUMENTS['ninja'] == '0':
+        return False
+    if len(COMMAND_LINE_TARGETS) != 1:
+        raise Exception('Can only generate a ninja file for a single target')
+    cl_target = COMMAND_LINE_TARGETS[0]
+    if 'vcxproj' in cl_target:
+        raise Exception('Cannot generate a ninja file for a vcxproj')
+    s = cl_target.split('.')
+    if ( style == 'unity' and 'nounity' in s or
```

Comment:
@rec: OK, I'll put in the effort to make ninja a target this weekend. FYI: `ninja` is not part of `ARGUMENTS` unless it has an assignment, so `ninja not in ARGUMENTS` isn't going to work. We also get an error about `do not know how to make target "ninja"`. I don't know scons that well, but I'll figure it out.